test/logic/LocationTest.java

```diff
@@ -0,0 +1,240 @@
+package logic;
+
+import models.*;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+
+public class LocationTest {
+    @Test
+    public void createLocation() {
+        // Arrange
+        String name = "location";
+        Guard invariant = new TrueGuard();
+        boolean isInitial = false;
+        boolean isUrgent = false;
+        boolean isUniversal = false;
+        boolean isInconsistent = false;
+        int x = 0;
+        int y = 0;
+        List<Location> children = new ArrayList<>();
+
+        // Act
+        Location location = Location.create(
+                name, invariant, isInitial, isUrgent, isUniversal, isInconsistent, x, y
+        );
+
+        // Assert
+        assertEquals(location.getName(), name);
+        assertEquals(location.isInitial(), isInitial);
+        assertEquals(location.isUrgent(), isUrgent);
+        assertEquals(location.isUniversal(), isUniversal);
+        assertEquals(location.isInconsistent(), isInconsistent);
+        assertEquals(location.getX(), x);
+        assertEquals(location.getY(), y);
+        assertEquals(location.getChildren(), children);
+    }
+
+    @Test
+    public void createLocationDefaultsCoordinatesToZeroZero() {
+        // Arrange
+        String name = "location";
+        Guard invariant = new TrueGuard();
+        boolean isInitial = false;
+        boolean isUrgent = false;
+        boolean isUniversal = false;
+        boolean isInconsistent = false;
+        int x = 0;
+        int y = 0;
+        List<Location> children = new ArrayList<>();
+
+        // Act
+        Location location = Location.create(
+                name, invariant, isInitial, isUrgent, isUniversal, isInconsistent
+        );
+
+        // Assert
+        assertEquals(location.getName(), name);
+        assertEquals(location.isInitial(), isInitial);
+        assertEquals(location.isUrgent(), isUrgent);
+        assertEquals(location.isUniversal(), isUniversal);
+        assertEquals(location.isInconsistent(), isInconsistent);
+        assertEquals(location.getX(), x);
+        assertEquals(location.getY(), y);
+        assertEquals(location.getChildren(), children);
+    }
+
+    @Test
+    public void createInitialLocationDefaultsCoordinatesToZeroZero() {
+        // Arrange
+        String name = "initial location";
```

Comment:
I would personally not allow spaces in names of model entities.

Reply:
I have opened #98 to do this.